tests/Test.ADAL.NET.Unit/WsTrustTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.WsTrust;
+using System.IO;
+using System.Xml.Linq;
+
+namespace Test.ADAL.NET.Unit
+{
+    [TestClass]
+    [DeploymentItem("WsTrustResponse.xml")]
+    public class WsTrustTests
+    {
+        [TestMethod]
+        [TestCategory("WsTrustTests")]
+        public void TestCreateFromResponseDocument_WhenInputContainsWhitespace_ShouldPreserveWhitespace()
+        {
+            string sample = File.ReadAllText("WsTrustResponse.xml");
+            string characteristic = "\n        <saml:Assertion";
+            StringAssert.Contains(sample, characteristic);
+            WsTrustResponse resp = WsTrustResponse.CreateFromResponseDocument(
+                XDocument.Parse(sample, LoadOptions.PreserveWhitespace), WsTrustVersion.WsTrust2005);
```

Comment:
You can invoke and test ReadDocumentFromResponse as well to read the actual file instead of doing XDocument.Parse

Reply:
OK I see your point on also testing the changed method `ReadDocumentFromResponse()`. The upcoming new commit will do it, but in a slightly different way. Instead of testing both `ReadDocumentFromResponse()` and `CreateFromResponseDocument()` in one test case, a new test case will be added to test the higher level `CreateFromResponse()` which invokes both lower level helpers. Hope that helps.